lib/xcodeproj/project/object/helpers/file_references_factory.rb

```diff
@@ -126,7 +126,7 @@ def new_file_reference(group, path, source_tree)
           # @return [XCVersionGroup] The new reference.
           #
           def new_xcdatamodeld(group, path, source_tree)
-            path = Pathname.new(path)
+            path = Pathname.new(group.parent.path).realpath().join(group.path).join(path)
```

Comment:
group.path doesn't have a method realpath since group.path is just a string hence the call to Pathname.new:
```
Pathname.new(group.path).realpath
```
group.parent.path is necessary because group.path returns No such file or directory when we call realpath.